tamr_client/datasets/dataset.py

```diff
@@ -85,6 +87,7 @@ def _from_json(url: tc.URL, data: JsonDict) -> Dataset:
         name=cp["name"],
         description=cp.get("description"),
         key_attribute_names=tuple(cp["keyAttributeNames"]),
+        version = cp.get("version"),
```

Comment:
I believe that `version` is a required field for datasets, so avoid `.get(...)` since connotes to readers that `version` field might not be present in the JSON (and that `version` property might be set to `None` in the `Dataset`)

```suggestion
        version = cp["version"],
```